### PR TITLE
Feature/database

### PR DESCRIPTION
**Pull request by gunnarmon**
Connect mongoDB

### CODE REVIEW

File: components/RecipeForm/RecipeForm.tsx [hunk 1]
```diff
@@ -186,7 +185,7 @@ export default function RecipeForm({
 
     const newRecipe = {
       ...recipeData,
-      id: crypto.randomUUID(),
+      // id: crypto.randomUUID(),
```
Comment (NikRoe) [inline components/RecipeForm/RecipeForm.tsx]: if you don't need this anymore, you can remove it

```suggestion
```

File: lib/units.json [hunk 2]
```diff
@@ -7,6 +7,6 @@
   { "id": "6", "unit": "tsp" },
   { "id": "7", "unit": "tbsp" },
   { "id": "8", "unit": "piece(s)" },
-  { "id": "9", "unit": "cub" },
+  { "id": "9", "unit": "cup" },
```
Comment (NikRoe) [inline lib/units.json:10]: 🐻

File: types.ts [hunk 3]
```diff
@@ -5,7 +5,8 @@ export type Recipe = {
   cookingTime: number;
   description: string;
   difficulty: string;
-  id: string;
+  _id?: string;
+  id?: string;
```
Comment (NikRoe) [inline types.ts]: can there be either or?

File: pages/api/recipes/index.js
Comment (NikRoe) [inline pages/api/recipes/index.js:1]: Should this file not be a `.ts` file?

File: pages/api/recipes/[id].js
Comment (NikRoe) [inline pages/api/recipes/[id].js:1]: Should this file not be a `.ts` file?

File: db/connect.js
Comment (NikRoe) [inline db/connect.js:1]: Should this file not be a `.ts` file?

File: db/models/Recipe.js
Comment (NikRoe) [inline db/models/Recipe.js:1]: Should this file not be a `.ts` file?

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi there,

nice work!

I left a few comments since I believe you mistakenly gave several files a `.js` ending instead of a `.ts` ending. This will most likely force you to do some more typing.

I also have a question that concerns the functions in the `_app.tsx`.

Currently you fetch the data in the `_app.tsx` and write all functions that update what is in the database within the same file.
This is an approach we typically choose when we have local states in a Next application and want those states and functions to be available to more than one page.

If you need some of the functions, for instance the `handleDeleteRecipe` function only within a certain page, it makes more sense to fetch your data there as well (due to the caching of `useSWR` this is still performant) and also write the function `handleDeleteRecipe` in that location.

That way the function would only be created where you actually need it and not regardless of the page you are currently on.

So the question is: Do you need all of the functions that you wrote in the `_app.tsx` in several pages? 
If so, leave them as they are.
If not, move the affected function to where it is actually needed.

Keep it up 🙌

File: components/RecipeForm/RecipeForm.tsx (hunk 1 above)
Comment (gunnarmon) [reply components/RecipeForm/RecipeForm.tsx]: done. thanks!

File: types.ts (hunk 3 above)
Comment (gunnarmon) [reply types.ts]: thanks

**Comment by stekug** [comment]
We have adopted your suggestions and changed our code. Thanks for your input!

**Review by Roland-Hufnagel — APPROVED** [review]
LGTM

**Pull request merged**